src/coreclr/src/vm/classlayoutinfo.cpp

```diff
@@ -0,0 +1,1120 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "common.h"
+#include "class.h"
+#include "fieldmarshaler.h"
+
+#ifndef DACCESS_COMPILE
```

Comment:
This seems to be all cut&paste from the other file. Any interesting deltas worth looking at?

Reply:
Looking through the general logic of the `CollectLayoutFieldMetadataThrowing` and `CollectNativeLayoutFieldMetadataThrowing` would be good. Also checking over the implementation of `InitializeNativeLayoutFieldMetadataThrowing` to make sure I got the thread-safety right this time.